DylibBinder/XmlGenerator.cs

```diff
@@ -0,0 +1,258 @@
+										   ("isMutating", func.IsMutating), ("operatorKind", func.OperatorKind),
+										   ("accessibility", func.Accessibility), ("isDeprecated", func.IsDeprecated),
+										   ("isUnavailable", func.IsUnavailable), ("isFinal", func.IsFinal),
+										   ("isOptional", func.IsOptional), ("isRequired", func.IsRequired),
+										   ("isConvenienceInit", func.IsConvenienceInit), ("objcSelector", func.ObjcSelector),
+										   ("isPossiblyIncomplete", func.IsPossiblyIncomplete));
+
+			WriteParameterLists (writer, func.ParameterLists);
+			WriteGenericParameters (writer, func.GenericParameters);
+			writer.WriteEndElement ();
+		}
+
+		void WriteParameterLists (XmlWriter writer, DBParameterLists parameterLists)
+		{
+			writer.WriteStartElement ("parameterlists");
+			foreach (var parameterList in parameterLists.ParameterLists) {
+				WriteParameterList (writer, parameterList);
+			}
+			writer.WriteEndElement ();
+		}
+
+		void WriteParameterList (XmlWriter writer, DBParameterList parameterList)
+		{
+			writer.WriteStartElement ("parameterlist");
+			writer.WriteAttributeString ("index", parameterList.Index);
+			foreach (var parameter in parameterList.Parameters) {
+				WriteParameter (writer, parameter);
+			}
+			writer.WriteEndElement ();
+		}
+
+		void WriteParameter (XmlWriter writer, DBParameter parameter)
+		{
+			if (parameter.IsEmptyParameter)
+				return;
+
+			writer.WriteStartElement ("parameter");
+			WriteMultipleAttributeStrings (writer, ("index", parameter.Index),
+										   ("type", parameter.Type), ("publicName", parameter.PublicName),
+										   ("privateName", parameter.PrivateName), ("isVariadic", parameter.IsVariadic));
+			writer.WriteEndElement ();
+		}
+
+		void WriteProperties (XmlWriter writer, DBTypeDeclaration typeDeclaration)
+		{
+			foreach (var property in typeDeclaration.Properties.Properties) {
+				WriteProp (writer, property);
+				WriteFunc (writer, property.Getter);
+				if (property.Setter != null)
+					WriteFunc (writer, property.Setter);
+			}
+		}
+
+		void WriteProp (XmlWriter writer, DBProperty propery)
+		{
+			writer.WriteStartElement ("property");
+			WriteMultipleAttributeStrings (writer, ("name", propery.Name),
+										   ("type", propery.Type), ("isStatic", propery.IsStatic),
+										   ("accessibility", propery.Accessibility), ("isDeprecated", propery.IsDeprecated),
+										   ("isUnavailable", propery.IsUnavailable), ("isOptional", propery.IsOptional),
+										   ("storage", propery.Storage), ("isPossiblyIncomplete", propery.IsPossiblyIncomplete));
+			writer.WriteEndElement ();
+		}
+
+		void CloseWriter (XmlWriter writer) {
+			writer.WriteEndDocument ();
+			writer.Close ();
+		}
+
+		void WriteMultipleAttributeStrings (XmlWriter writer, params (string name, string value) [] attributes)
+		{
+			foreach (var attribute in attributes) {
+				writer.WriteAttributeString (attribute.name, attribute.value);
+			}
+		}
+
+	}
+}
```

Comment:
After looking at the full class. we only have an instance variable that is the writer, we could do inheritance and avoid the need to implement IDIsposable which is always an annoying interface to implement.

Reply:
Discussion